ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala

```diff
@@ -77,38 +80,73 @@ object LedgerClient {
     }
   }
 
-  def singleHost(hostIp: String, port: Int, configuration: LedgerClientConfiguration)(
-      implicit ec: ExecutionContext,
-      esf: ExecutionSequencerFactory): Future[LedgerClient] = {
-
-    val builder: NettyChannelBuilder = NettyChannelBuilder
-      .forAddress(hostIp, port)
-
+  /**
+    * Constructs a [[Channel]], ensuring that the [[LedgerClientConfiguration]] is picked up and valid
+    *
+    * You'll generally want to use [[singleHost]], use this only if you need a higher level of control
+    * over your [[Channel]].
+    */
+  def constructChannel(
+      hostIp: String,
+      port: Int,
+      configuration: LedgerClientConfiguration): Try[Channel] = {
+    val builder: NettyChannelBuilder = NettyChannelBuilder.forAddress(hostIp, port)
     configuration.sslContext
       .fold {
-        builder.usePlaintext()
+        if (configuration.accessToken.isDefined) {
+          Failure(new IllegalArgumentException("Cannot send a token over an insecure channel"))
```

Comment:
Should we provide a way to use tokens on insecure channels for development purposes?

I like the strict check here, provided that we have an easy way of working with a local sandbox in SSL mode (easy meaning the user does not have to manually run SSL certificate generating commands).

Reply:
I see you added some certificates in `navigator/backend/scenarios/rental/tls/` :+1:

Reply:
As @gerolf-da mentioned offline, we should probably not do it here. The thing I was mostly interested in was providing a clear error message to the user. I'll see what happens if I try to do it without any such protection. IIRC gRPC itself should complain, I'll try to see if it's possible to get back some meaningful error to return to the user.